File: src/Enterprise/C#/Example3.cs
using System;

// Abstract Product
public interface Report {
    void Generate();
}

// Concrete Product
public class PDFReport : Report {
    public void Generate() {
        Console.WriteLine("Generating PDF report");
    }
}

public class HTMLReport : Report {
    public void Generate() {
        Console.WriteLine("Generating HTML report");
    }
}

// Abstract Factory
public interface ReportFactory {
    Report CreateReport();
}

// Concrete Factory
public class PDFReportFactory : ReportFactory {
    public Report CreateReport() {
        return new PDFReport();
    }
}

public class HTMLReportFactory : ReportFactory {
    public Report CreateReport() {
        return new HTMLReport();
    }
}

public class Example3 {
    public static void UseFactory(ReportFactory factory) {
        var report = factory.CreateReport();
        report.Generate();
    }

    public static void Main(string[] args) {
        UseFactory(new PDFReportFactory());
        UseFactory(new HTMLReportFactory());
    }
}

File: src/Enterprise/Java/Example3.java
// Abstract Product
interface Report {
    void generate();
}

// Concrete Product
class PDFReport implements Report {
    @Override
    public void generate() {
        System.out.println("Generating PDF report");
    }
}

class HTMLReport implements Report {
    @Override
    public void generate() {
        System.out.println("Generating HTML report");
    }
}

// Abstract Factory
interface ReportFactory {
    Report createReport();
}

// Concrete Factory
class PDFReportFactory implements ReportFactory {
    @Override
    public Report createReport() {
        return new PDFReport();
    }
}

class HTMLReportFactory implements ReportFactory {
    @Override
    public Report createReport() {
        return new HTMLReport();
    }
}

public class Example3 {
    public static void useFactory(ReportFactory factory) {
        Report report = factory.createReport();
        report.generate();
    }

    public static void main(String[] args) {
        useFactory(new PDFReportFactory());
        useFactory(new HTMLReportFactory());
    }
}
